python-backend/app/api/permits/permit/resources/permit.py

```diff
@@ -1,15 +1,102 @@
-from flask_restplus import Resource
+from flask_restplus import Resource, reqparse
+from datetime import datetime
+from flask import current_app, request
+
 from ..models.permit import Permit
+from ...permit_amendment.models.permit_amendment import PermitAmendment
+from ....mines.mine.models.mine import Mine
 from app.extensions import api
-from ....utils.access_decorators import requires_role_mine_view
+from ....utils.access_decorators import requires_role_mine_view, requires_role_mine_create
 from ....utils.resources_mixins import UserMixin, ErrorMixin
 
 
 class PermitResource(Resource, UserMixin, ErrorMixin):
+
+    parser = reqparse.RequestParser()
+    parser.add_argument('permit_no', type=str, help='Number of the permit being added.')
+    parser.add_argument('mine_guid', type=str, help='guid of the mine.')
+    parser.add_argument(
+        'permit_status_code',
+        type=str,
+        help='Status of the permit being added.',
+        store_missing=False)
+    parser.add_argument(
+        'received_date', type=lambda x: datetime.strptime(x, '%Y-%m-%d') if x else None)
+    parser.add_argument(
+        'issue_date', type=lambda x: datetime.strptime(x, '%Y-%m-%d') if x else None)
+    parser.add_argument(
+        'authorization_end_date', type=lambda x: datetime.strptime(x, '%Y-%m-%d') if x else None)
+    parser.add_argument(
+        'permit_amendment_status_code', type=str, help='Status of the permit being added.')
+
     @api.doc(params={'permit_guid': 'Permit guid.'})
     @requires_role_mine_view
-    def get(self, permit_guid):
+    def get(self, permit_guid=None):
+        permit_no = request.args.get('permit-no', None, type=str)
+
+        if permit_no:
+            permit = Permit.find_by_permit_no(permit_no)
+            if permit:
+                return self.create_error_payload(
+                    400, 'There was a permit found with the provided permit number.'), 400
```

Comment:
error string is missing a 'not'

Reply:
No it's not, this is a fail out if it finds a permit because that means the permit number is already in use.

Reply:
Ah, how about 'That permit number is already in use'? to indicate that they need to change it?

Reply:
Changed both

Reply:
After discussion, this is the GET method, it shouldn't be returning an error all. if searched by permit_no, just return it. The caller may have logic, but this method shouldn't.